lib/react/jsx/processor.rb

```diff
@@ -0,0 +1,17 @@
+module React
+  module JSX
+    class Processor
+      def self.instance
+        @instance ||= new
+      end
+      
+      def self.call(input)
+        instance.call(input)
+      end
+
+      def call(input)
+        @output ||= JSX::transform(input[:data])
```

Comment:
I think this was done for performance reasons in the Tilt template version of this. For these Sprocket processors there is a way to do a similar thing by implementing an optional `cache_key` method.